cmd/security-secretstore-setup/main.go

```diff
@@ -73,7 +74,21 @@ func main() {
 	//step 3: initialize and unseal Vault
 
 	//Step 4:
-	//TODO: create vault access token for different roles
+	ctx, cancelTokenProvider := context.WithCancel(context.Background())
+	defer cancelTokenProvider()
+	tokenProvider := secretstore.NewTokenProvider(ctx)
+	if secretstore.Configuration.SecretService.TokenProvider != "" {
+		if err := tokenProvider.SetConfiguration(secretstore.Configuration.SecretService); err != nil {
+			secretstore.LoggingClient.Error("failed to configure token provider")
+			os.Exit(1)
+		}
+		if err := tokenProvider.Launch(); err != nil {
+			secretstore.LoggingClient.Error("token provider failed")
```

Comment:
Should `err` also be part of error log message? or at least included in debug level for better trouble-shooting.